monai/apps/deepedit/transforms.py

```diff
@@ -0,0 +1,170 @@
+# Copyright 2020 - 2021 MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+import logging
+from typing import Dict, Tuple
+
+import numpy as np
+
+from monai.transforms.transform import Randomizable, Transform
+
+logger = logging.getLogger(__name__)
+
+from monai.utils import optional_import
+
+distance_transform_cdt, _ = optional_import("scipy.ndimage.morphology", name="distance_transform_cdt")
+
+
+class DiscardAddGuidanced(Transform):
```

Comment:
I think all the dict-based transforms should inherit `MapTransform` base class and use `keys`.

Thanks.

Reply:
Many thanks for the comment @Nic-Ma. Following your recommendation, I modified one of the transforms that can take advantage of inheriting from _MapTransform_. The other two transforms are ad hoc transforms to deepedit. Similar to what has been done to the deepgrow transforms.